docs/best_practice/systables.txt

```diff
@@ -0,0 +1,161 @@
+    ...
+
+lists all available attributes. This object has the same structure as
+the **_node** attribute of **sys.operations** from the previous
+section.
+
+Step 4: Insights about Partitions, Shards, and Replication
+==========================================================
+
+Crate divides the rows of each table into shards that are distinctively
+distributed to all nodes in your cluster. Replication uses the same
+mechanism to add redundancy and thus resilience to your data. While most
+of the time Crate takes transparently care of distributing and
+replicating the shards, it’s useful during troubleshooting to actually
+find out some more about these data structures. The **sys.shards** table
+provides access to the status and size of shards, its names and IDs::
+
+    cr> SHOW COLUMNS IN shards FROM sys;
+    ...
+
+The cluster state is somewhat delicate when nodes join or leave, since
+in those situations shards have to be rearranged to ensure that each of
+them is replicated to different nodes. As long as the **state**
+attribute is ‘STARTED’ for all shards, the cluster is in a stable state,
+otherwise Crate is occupied with some background activity. The cluster
+state indicators on the admin UI evaluate these values as well.
+
+The **sys.shards** table contains even more information about the
+rebalancing activities. Whenever Crate needs to copy one shard to
+another node -- since that may be necessary to ensure there are enough
+replica of it distributed in the cluster -- you can monitor the progress
+of that operation with the **recovery** object::
+
+    cr> SELECT DISTINCT partition_ident, recovery['size']['percent'], recovery['total_time'] FROM sys.shards WHERE primary;
+    ...
+
+To find out on which specific node a shard is stored, also use the
+_node object that is available for this table. As an example::
+
+    cr> SELECT _node['hostname'], table_name, num_docs FROM sys.shards ORDER BY num_docs DESC LIMIT 3;
+    ...
+
+list the hosts and tables with the highest number of rows inside a
+single shard. There’s a lot of additional information available for
+shards. To find out more about that, see our documentation.
+
+Step 5: Managing Snapshots
+==========================
+
+Finally, if your repair efforts did not succeed, your application or
+users accidently deleted some data, recover one of the previous taken
+snapshot of your cluster. The tables **sys.snapshots** and
+**sys.repositories** assist you in managing your backups. Remember, one
+or more backups are stored in repositories outside the Crate cluster
+initialized with the **CREATE REPOSITORY** request. An actual copy of a
+current database state is made with the **CREATE SNAPSHOT** command. If
+you forgot where you store your snapshots::
+
+    cr> SELECT * FROM sys.repositories;
+    ...
+
+might come in handy. To actually recover data, first determine which
+snapshot to roll back. Suppose you make nightly backups, the command::
```

Comment:
i would not use "roll back"  here since it is used for transactions etc ... probably "restore"?